ros/src/simulation/gazebo_simulator/vehicle/vehicle_gazebo_simulation_interface/src/vehicle_gazebo_info_publisher.cpp

```diff
@@ -0,0 +1,102 @@
+#include <ros/ros.h>
+#include <std_msgs/Float64.h>
+#include <geometry_msgs/PoseStamped.h>
+#include <gazebo_msgs/GetLinkState.h>
+#include <sensor_msgs/JointState.h>
+#include <autoware_msgs/VehicleStatus.h>
+#include <string>
+#include <cmath>
+
+class VehicleGazeboInfoPublisher
+{
+  private:
+    ros::NodeHandle nh_;
+    ros::NodeHandle pnh_;
+    ros::ServiceClient client_;
+    ros::Publisher vehicle_pose_pub_;
+    ros::Publisher vehicle_vel_pub_;
+    ros::Publisher steering_angle_pub_;
+    ros::Publisher vehicle_status_pub_;
+    ros::Timer publish_timer_; // publish timer
+    ros::Subscriber odom_sub_;
+    double wheel_radius_;
+    std::string ns_;
+    void publishTimerCallback(const ros::TimerEvent &e);
+    void jointStateCallback(const sensor_msgs::JointState::ConstPtr &input_msg);
+
+  public:
+    VehicleGazeboInfoPublisher();
+    ~VehicleGazeboInfoPublisher(){};
+};
+
+VehicleGazeboInfoPublisher::VehicleGazeboInfoPublisher() : nh_(""), pnh_("~")
+{
+    client_ = nh_.serviceClient<gazebo_msgs::GetLinkState>("/gazebo/get_link_state");
+    vehicle_pose_pub_ = nh_.advertise<geometry_msgs::PoseStamped>("/vehicle_info/pose", 1, true);
+    vehicle_vel_pub_ = nh_.advertise<std_msgs::Float64>("/vehicle_info/velocity", 1, true);
+    steering_angle_pub_ = nh_.advertise<std_msgs::Float64>("/vehicle_info/steering_angle", 1, true);
+    vehicle_status_pub_ = nh_.advertise<autoware_msgs::VehicleStatus>("/vehicle_status", 1, true);
+    double publish_pose_rate;
+    pnh_.param<double>("publish_pose_rate", publish_pose_rate, double(10.0));
+    pnh_.param<double>("wheel_radius", wheel_radius_, 0.341);
+    pnh_.param("ns", ns_, std::string("autoware_gazebo"));
+    publish_timer_ = nh_.createTimer(ros::Duration(1.0 / publish_pose_rate), &VehicleGazeboInfoPublisher::publishTimerCallback, this);
+    odom_sub_ = nh_.subscribe("joint_states", 1, &VehicleGazeboInfoPublisher::jointStateCallback, this);
+}
+
+void VehicleGazeboInfoPublisher::publishTimerCallback(const ros::TimerEvent &e)
+{
+    gazebo_msgs::GetLinkState base_link_srv;
+    base_link_srv.request.link_name = ns_ + "::base_link";
```

Comment:
`ns_` is needed, right?

Reply:
Since the link of gazebo requires ns, it is the intended description

Reply:
ok